src/Microsoft.OData.Core/Json/JsonValueUtils.cs

```diff
@@ -51,7 +52,8 @@ internal static class JsonValueUtils
         /// <summary>
         /// Map of special characters to strings.
         /// </summary>
-        private static readonly string[] SpecialCharToEscapedStringMap = JsonValueUtils.CreateSpecialCharToEscapedStringMap();
+        private static Dictionary<char, string> SpecialCharToEscapedStringMap = new Dictionary<char, string>() { { '\r', "\\r" }, { '\t', "\\t" }, { '\n', "\\n" }, 

```

Comment:
>Dictionary [](start = 23, length = 10)

This is not threadsafe, as you are adding to the dictionary in line 96. Probably have to make this a concurrentdictionary. I'd want to see perf comparisons to validate that looking up/lazily adding to a concurrent dictionary would be more performant than constructing the map up-front and doing a simple array index look-up.  For frequently lookups (where performance becomes more critical) I would think that the readonly string[] would probably be faster.